### PR TITLE
chore: Fix test builds with meck incompatibilities.

### PR DESCRIPTION
**Pull request by kinyoklion**
Works around: https://github.com/eproxus/meck/issues/248

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -15,7 +15,7 @@
         {deps, [
             {bookish_spork, "0.3.5"},
             {cowboy, "2.8.0"},
-            {meck, "0.9.2"}
+            {meck, { git, "git@github.com:eproxus/meck.git", {ref, "dfb47c544bdf14b5c26ff2651cf9339dbbe2f50e"}}}
```
Comment (kinyoklion) [inline rebar.config:18]: This SHA has some OTP/27 fixes.

File: Makefile [hunk 2]
```diff
@@ -24,7 +24,7 @@ doc:
 
 tests:
 	docker run --name ld-test-redis -p 6379:6379 -d redis
-	@$(REBAR3) ct --dir="test,test-redis" --logdir logs/ct --cover
+	@$(REBAR3) ct --dir="test,test-redis" --logdir logs/ct
```
Comment (kinyoklion) [inline Makefile:27]: Removing cover here is enough to make things work for dev in OTP 25+.

File: Makefile [hunk 3]
```diff
@@ -33,7 +33,14 @@ release-tests:
 
 #This is used on CircleCI because the Redis Docker container is already started unlike the local tests command
 ci-tests:
-	@$(REBAR3) ct --dir="test,test-redis" --logdir logs/ct
+	@if [ "$(ERL_VERSION)" -ge "25" ]; then\
```
Comment (kinyoklion) [inline Makefile:36]: Here things get ugly. Need two different versions of meck to keep things working.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -18,7 +18,6 @@ jobs:
       fail-fast: false
       matrix:
         versions:
-        - {os: 'ubuntu-20.04', otp: '21.x', rebar: '3.15.2'}
```
Comment (kinyoklion) [inline .github/workflows/ci.yml:21]: I'm having to drop CI for 21.

Comment (kinyoklion) [reply .github/workflows/ci.yml:21]: It should all still work, but a version of meck that works with 21 will not work with 27.

File: src/ldclient_backoff.erl
Comment (kinyoklion) [inline src/ldclient_backoff.erl:1]: Modified the backoff functionality to allow injecting a rand function. meck seems especially unhappy with mocking rand in OTP 27.

File: Makefile (hunk 2 above)
Comment (kinyoklion) [inline Makefile:27]: First problem is that it doesn't work with the cover arguments in current OTP 27.

**Review by keelerm84 — APPROVED** [review]

**Pull request merged**